cdap-elastic/src/main/java/io/cdap/cdap/metadata/elastic/ElasticsearchMetadataStorage.java

```diff
@@ -1010,21 +1010,31 @@ private QueryBuilder createMainQuery(SearchRequest request) {
     // all terms must occur in the text field as selected by the scope in the search request.
     String textField = request.getScope() == null ? TEXT_FIELD : request.getScope().name().toLowerCase();
 
-    // split the query into its terms and iterate over all terms
-    Iterable<String> terms = Splitter.on(SPACE_SEPARATOR_PATTERN)
-      .omitEmptyStrings().trimResults().split(request.getQuery());
-    List<QueryBuilder> termQueries = new ArrayList<>();
-    for (String term : terms) {
-      termQueries.add(createTermQuery(term, textField, request));
+    // split the query into its parsed terms and create corresponding QueryBuilders for each
+    Map<QueryBuilder, QueryTerm> termQueries = new HashMap<>();
+    for (QueryTerm queryTerm : QueryParser.parse(request.getQuery())) {
+      termQueries.put(createTermQuery(queryTerm.getTerm(), textField, request), queryTerm);
     }
     if (termQueries.isEmpty()) {
       return QueryBuilders.matchAllQuery();
     }
     if (termQueries.size() == 1) {
-      return termQueries.get(0);
+      Map.Entry<QueryBuilder, QueryTerm> onlyEntry = termQueries.entrySet().iterator().next();
+      return onlyEntry.getKey();
+    }
+    // populate boolQuery clauses based on corresponding QueryTerm qualifier
+    final BoolQueryBuilder boolQuery = QueryBuilders.boolQuery();
```

Comment:
remove `final`. It is not needed in Java 8

Reply:
right, this must be a relic from when Java 7 was supported, that was refactored

Reply:
Addressed in the latest commit.